WordPress/Classes/Models/AbstractPost.swift

```diff
@@ -1,10 +1,81 @@
 import Foundation
 
 extension AbstractPost {
+
+    // MARK: - Status
+
+    @objc
+    var statusTitle: String? {
+        guard let status = self.status else {
+            return nil
+        }
+
+        return AbstractPost.title(for: status)
+    }
+
+    @objc
+    var remoteStatus: AbstractPostRemoteStatus {
+        get {
+            guard let remoteStatusNumber = remoteStatusNumber?.uintValue,
+                let status = AbstractPostRemoteStatus(rawValue: remoteStatusNumber) else {
+                    return .pushing
+            }
+
+            return status
+        }
+
+        set {
+            remoteStatusNumber = NSNumber(value: newValue.rawValue)
+        }
+    }
+
     class func title(for status: Status) -> String {
-        return AbstractPost.title(forStatus: status.rawValue)
+        return title(forStatus: status.rawValue)
```

Comment:
This is being called externally so it can't really be private.

Same question about `static` as above.

Reply:
That's weird. I changed this to `private` and the project still compiled. 

```swift 
    private static func title(for status: Status) -> String {
        return title(forStatus: status.rawValue)
    }
```

¯\\\_(ツ)\_/¯ 

Anyways, this is a special case. And I guess we do want it to be public since it's a utility function hiding inside a class.

Reply:
By "this method" I imagine you refer to the method including the referenced line.  I assumed you meant the method called on the referenced line.

The whole idea of having this method to begin with is to promote the usage of the `Status` enum over the raw value string.  Making it private would indeed defeat that purpose.

Reply:
> Making it private would indeed defeat that purpose.

Yes, I agree. I was wrong about suggesting to make this private. It's probably why I mentioned I was not sure if it should be in https://github.com/wordpress-mobile/WordPress-iOS/pull/12106#discussion_r303956800.

Reply:
In the case of promoting the usage of the `Status` enum, I think we should place the `title` resolver in the `enum` itself.

```swift
    enum Status: String {
        case draft = "draft"
        case pending = "pending"
        case publishPrivate = "private"
        case publish = "publish"
        case scheduled = "future"
        case trash = "trash"
        case deleted = "deleted" // Returned by wpcom REST API when a post is permanently deleted.
        
        var title: String {
            switch self {
            case .draft:
                return NSLocalizedString("Draft", comment: "Name for the status of a draft post.")
            case .pending:
                return NSLocalizedString("Pending review", comment: "Name for the status of a post pending review.")
            case .publishPrivate:
                return NSLocalizedString("Private", comment: "Name for the status of a post that is marked private.")
            case .publish:
                return NSLocalizedString("Published", comment: "Name for the status of a published post.")
            case .trash:
                return NSLocalizedString("Trashed", comment: "Name for the status of a trashed post")
            case .scheduled:
                return NSLocalizedString("Scheduled", comment: "Name for the status of a scheduled post")
            default:
                return rawValue
            }
        }
    }
```

Not today, but probably in the future.

Reply:
> By "this method" I imagine you refer to the method including the referenced line. I assumed you meant the method called on the referenced line.

Yes, sorry about that. 😞

Reply:
> In the case of promoting the usage of the Status enum, I think we should place the title resolver in the enum itself.

The scope of this PR is just to migrate the code, not to improve on it.  This is a deliberate choice  on my end to avoid coupling tasks that can be tackled separately.

Reply:
> The scope of this PR is just to migrate the code, not to improve on it. 

Right. I'm very sorry about that. I've gone way overboard on the review here. I didn't realize that you simply wanted to convert it to Swift without making any improvements since it adds to the development time. Again, my apologies. 😞

Reply:
Hah, no worries it's fine... it's not about the time only though - the thing is more about separating concerns and keeping the PR scoped for me.

I've pushed some PRs in the past which suffered a bit from scope creep due to tackling "just one more thing", and now I'm trying to get into the mindset of being very mindful of the initial scope, and making it clear in the PR.